climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+                    max_cons_id += 1
+                else:
+                    firms_cons.at[index, 'cons_id'] = firms_cons.at[(index-1), 'cons_id']
+
+            re_order = np.zeros(len(firms_cons), int)
+            for data, order in zip(firms_cons.cons_id.values, sort_idx):
+                re_order[order] = data
+            firms_iter.cons_id.values[firms_iter.event_id == event_id] = re_order
+
+        firms.cons_id.values[firms['iter_ev'].values] = firms_iter.cons_id.values
+        return firms
+
+    @staticmethod
+    def _firms_clustering(firms, res_data, clus_thres):
+        """ Compute geographic clusters and sort firms with ascending clus_id
+        for each cons_id. Geographic clusters are identified using sci-kit
+        learn's DBSCAN algorithm, which finds core samples of high density
+        and expands clusters from them.
+        
+        https://scikit-learn.org/stable/modules/generated/sklearn.cluster.DBSCAN.html
+
+        Parameters
+        ----------
+            firms : pd.DataFrame
+                 FIRMS data
+            res_data : float
+                FIRMS instrument resolution in degrees
+            clus_thres : int
+                Clustering factor which multiplies instrument resolution
+
+        Returns
+        -------
+            firms : pd.DataFrame
+                FIRMS data including info on spacial cluster per point
+
+        """
+
+        LOGGER.debug('Computing geographic clusters in consecutive fires.')
+        firms_iter = firms[firms['iter_ev']][['latitude', 'longitude', 'cons_id',
+                                              'clus_id', 'event_id']]
+
+        for event_id in np.unique(firms_iter.event_id.values):
+
+            firms_cons = firms_iter[firms_iter.event_id == event_id]
+
+            # Creation of an identifier for geographical clustering
+            # For each temporal cluster, perform geographical clustering with DBSCAN algo
+            for cons_id in np.unique(firms_cons['cons_id'].values):
+                temp = np.argwhere(firms_cons['cons_id'].values == cons_id).reshape(-1,)
+                lat_lon = firms_cons.iloc[temp][['latitude', 'longitude']].values
+                lat_lon_uni, lat_lon_cpy = np.unique(lat_lon, return_inverse=True, axis=0)
+                cluster_id = DBSCAN(eps=res_data*clus_thres, min_samples=1).\
+                                    fit(lat_lon_uni).labels_
+                cluster_id = cluster_id[lat_lon_cpy]
+                firms_cons.clus_id.values[temp] = cluster_id
+                firms_iter.clus_id.values[firms_iter.event_id == event_id] = \
+                    firms_cons.clus_id.values
+
+        firms.clus_id.values[firms['iter_ev'].values] = firms_iter.clus_id.values
+
+        return firms
+
+    @staticmethod
+    @numba.njit(parallel=True)
+    def _firms_fire(days_thres, fir_cons_id, fir_clus_id, fir_ev_id, fir_iter,
+                    fir_date):
```

Comment:
To complete Evelyns comment:

```suggestion
    @numba.njit(parallel=True)
    def _firms_fire(df_firms):
```